File: 백준/Silver/14490. 백대열/백대열.c
#include <stdio.h>

int gcd(a, b) {
  int r;
  while (b != 0) {
    r = a % b;
    a = b;
    b = r;
  }
  return a;
}

int main(void) { 
   int a, b;
   char k;
   scanf("%d%c%d", &a, &k, &b);

   int num = gcd(a, b);
   printf("%d:%d", a / num, b / num);
   
   return 0;
}

File: 백준/Silver/14490. 백대열/백대열.py
def gcd(a, b):
  while(b != 0):
    r = a % b
    a = b
    b = r
  
  return a

index = input()

a, b = map(int, index.split(":"))

num = gcd(a, b)

print(f"{a // num}:{b // num}")